### PR TITLE
Make the sample-keys program emit b58-formatted public address files as well

### PR DESCRIPTION
**Pull request by cbeck88**
This is helpful if you want to fund an account created using the
    `sample-keys` program. The main reason to do that is that the `sample-keys`
    program is easy to run in automation rather than having spreadsheets or
    tables of keys, and it produces the root-identity files which our targets like
    bootstrap, fog-distribution and the test-client are compatible with.
    
    However, the b58 format is what full-service and other wallets use,
    and it's not super easy to convert from the b58 format to this format.
    
    An alternative here is that we could make the `.pub` file be the b58
    format and kill the old proto format, but that's a breaking change
    and I'm not sure what parts of infrastructure are still using these
    files, so I'd rather not do that and just remove the `.pub` file
    entirely at some later date.

    This work was done in support of funding the new test client

### CODE REVIEW

File: util/keyfile/src/keygen.rs [hunk 1]
```diff
@@ -19,14 +19,14 @@ pub fn write_keyfiles<P: AsRef<Path>>(
...
 
     fs::create_dir_all(&path)?;
 
     write_keyfile(path.as_ref().join(name).with_extension("json"), root_id)?;
-    write_pubfile(
-        path.as_ref().join(name).with_extension("pub"),
-        &acct_key.default_subaddress(),
-    )?;
+    write_pubfile(path.as_ref().join(name).with_extension("pub"), &addr)?;
+    write_keyfile(path.as_ref().join(name).with_extension("json"), root_id)?;
```
Comment (eranrund) [inline util/keyfile/src/keygen.rs:28]: duplicate?

File: util/keyfile/src/lib.rs [hunk 2]
```diff
@@ -59,7 +61,50 @@ pub fn read_pubfile_data<R: std::io::Read>(
...
+/// Write user b58 public address to disk
+pub fn write_b58pubfile<P: AsRef<Path>>(
+    path: P,
+    addr: &PublicAddress,
+) -> Result<(), std::io::Error> {
+    let mut wrapper = PrintableWrapper::new();
+    wrapper.set_public_address(addr.into());
+
+    let data = wrapper.write_to_bytes().map_err(to_io_error)?;
+    let data = bs58::encode(data).into_vec();
```
Comment (eranrund) [inline util/keyfile/src/lib.rs:73]: Should use https://github.com/mobilecoinfoundation/mobilecoin/blob/master/api/src/display.rs#L43

File: util/keyfile/src/lib.rs [hunk 3]
```diff
@@ -59,7 +61,50 @@ pub fn read_pubfile_data<R: std::io::Read>(
...
+/// Read user b58 pubfile from any implementor of `Read`
+pub fn read_b58pubfile_data<R: std::io::Read>(
+    buffer: &mut R,
+) -> Result<PublicAddress, std::io::Error> {
+    let data = {
+        let mut data = Vec::new();
+        buffer.read_to_end(&mut data)?;
+        data
+    };
+    let data = bs58::decode(data).into_vec().map_err(to_io_error)?;
```
Comment (eranrund) [inline util/keyfile/src/lib.rs:93]: Should use https://github.com/mobilecoinfoundation/mobilecoin/blob/master/api/src/display.rs#L54

**Review by eranrund — CHANGES_REQUESTED** [review]

File: util/keyfile/src/lib.rs (hunk 2 above)
Comment (garbageslam) [reply util/keyfile/src/lib.rs:73]: thanks

**Review by wjuan-mob — APPROVED** [review]

**Pull request merged**